step-functions/step-functions-client-api/src/main/java/step/functions/execution/TokenLifecycleInterceptor.java

```diff
@@ -0,0 +1,7 @@
+package step.functions.execution;
+
+public interface TokenLifecycleInterceptor {
+    void onReturnTokenHandle(String tokenHandleId);
+
+    void onGetTokenHandle(String tokenHandleId) throws Exception;
```

Comment:
Javadoc could be useful here. Especially to mention that an exception would block the token retrieval

Reply:
Documented in TokenLifecycleInterceptor class.